[hardknott] CHANGELOG: Merge 8.X release information into hardknott

amstewart opened the pull request:
It's getting to be burdensome to cherry-pick changes between the hardknott and sumo mainlines, and also to their release branches. We missed integrating some 8.X release information into hardknott already, and I think it will be confusing for users to determine which changelog they should be looking at.

So I'd like to adopt a policy of only using the latest mainline (in this case, "hardknott") as the canonical CHANGELOG and having all other branches just reference the github URL for the main changelog, rather than keeping their own copies.

`https://github.com/ni/nilrt/blob/HEAD/CHANGELOG.md`

As a first step, this PR merges the missing 8.X release information into the hardknott changelog, and also adds release information for 8.13.

Next, I'll add a line to the top of the sumo changelogs to direct users to the *canonical* changelog.

amstewart committed:
CHANGELOG: update changelog for the 21.8 release.

Service for the 2022C1 release.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>
(cherry picked from commit 06c86d153f1d397dc6b1d3377be81b61e0956f86)
(cherry picked from commit bd6659d050064d0eabb70d08dab81f171dd3cfa3)
Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
CHANGELOG: add reference to a fix in linux.git:**/buildnipkg

Signed-off-by: Alex Stewart <alex.stewart@ni.com>
(cherry picked from commit 17137e20cd60ce64ea5ea69cd807b23083693d95)
(cherry picked from commit 244b4e27136e38c5ce8edf028cbeb4d1aa4e1af7)
Signed-off-by: Alex Stewart <alex.stewart@ni.com>

chaitu236 committed:
CHANGELOG: update changelog for the 22.5 release.

Service for the 2022C2 release.

Signed-off-by: Chaitanya Vadrevu <chaitanya.vadrevu@ni.com>
(cherry picked from commit ddd80ac93f023ace9b4c479a53ffbf6fb364cf47)
(cherry picked from commit 07e074f7fa60a9e9f398c34edfba739b95265c11)
Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
CHANGELOG: merge 8.X release information

Merge the changelog snippets for the 8.X releases into the 9.X
changelog.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
CHANGELOG: add 8.13 release information

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
CHANGELOG: reference evergreen changelog

Add a link to the evergreen, canonical changelog file on GitHub, so that
maintainers only have to worry about keeping a single copy of the
changelog up-to-date.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

mike-petersen-ni commented on CHANGELOG.md:
```diff
@@ -100,6 +102,45 @@ Branch: `nilrt/22.5/hardknott`
+
+----
+
+## 8.13
+
+Branch: `nilrt/22.8/sumo`
+
```
Would we want a summary here like for 8.12 ("The 8.13 release is ...")?

amstewart replied:
Will fix.

amstewart committed:
CHANGELOG: add description for the 8.13 release

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart commented:
# Patch V2
* Rebased on latest `nilrt/master/hardknott` ref to pull other changelog updates.
* Added a short description to the 8.13 release - as we did with the 8.12 release. @mike-petersen-ni
